### PR TITLE
Configure references to add to unpublish confirmation context

### PR DESCRIPTION
**Pull request by monikasulik**

### CODE REVIEW

File: djangocms_references/cms_config.py [hunk 1]
```diff
@@ -110,11 +117,32 @@ def version_queryset_modifier(queryset):
     return queryset
 
 
+def unpublish_dependencies(request, version, *args, **kwargs):
+    """Render a partial template with a list of unpublish dependencies"""
+    references = get_all_reference_objects(
+        version.content, draft_and_published=True)
```
Comment (geoidesic) [inline djangocms_references/cms_config.py]: Why "draft_and_published"? Seems counter-intuitive... as in we're searching for unpublished items not items that are either draft or published.

File: djangocms_references/cms_config.py [hunk 2]
```diff
@@ -110,11 +117,32 @@ def version_queryset_modifier(queryset):
...
+    return render_to_string(
+        'djangocms_references/unpublish_dependencies.html',
+        {'references': references, 'extra_columns': get_extra_columns()}
+    )
+
+
 class ReferencesCMSAppConfig(CMSAppConfig):
     djangocms_references_enabled = True
+    djangocms_versioning_enabled = getattr(
+        settings, "DJANGOCMS_REFERENCES_VERSIONING_ENABLED", True
```
Comment (geoidesic) [inline djangocms_references/cms_config.py]: This check doesn't seem relevant to the PR title? Or are we doing more than just configuring references for Unpublish? In other words... why didn't we need to know if versioning was enabled for previous references functionality?

File: djangocms_references/templates/djangocms_references/unpublish_dependencies.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{% load i18n djangocms_references_tags %}
```
Comment (chaosk) [inline djangocms_references/templates/djangocms_references/unpublish_dependencies.html]: Perhaps we can reuse the template used for "Show references" view. Split the table to a separate template and `{% include %}` it in both "Show references" and this template?

Comment (monikasulik) [reply djangocms_references/templates/djangocms_references/unpublish_dependencies.html]: @chaosk I'm not sure we can... Is it definitely ok for the Show references one to use the same "No related objects" logic?

Comment (monikasulik) [reply djangocms_references/templates/djangocms_references/unpublish_dependencies.html]: Actually hmmmm... now that I've refactored the function the way I have, we probably could... ok I'll change it.

File: djangocms_references/cms_config.py (hunk 1 above)
Comment (monikasulik) [reply djangocms_references/cms_config.py]: We're actually not searching for unpublished items. We're searching for items that have relationships with an item we're unpublishing.

File: djangocms_references/cms_config.py (hunk 2 above)
Comment (monikasulik) [reply djangocms_references/cms_config.py]: We are configuring references for Unpublish and the unpublish functionality comes from versioning. Therefore before versioning could be happily disabled and references didn't need to know it even existed. But now that it knows about unpublishing, it needs to know about versioning and needs to tell versioning how to display the dependencies.
